diff --git a/lib/hash.c b/lib/hash.c
index 7d76d45..6f3a5c6 100644
--- a/lib/hash.c
+++ b/lib/hash.c
@@ -1,7 +1,7 @@
 /* hash - hashing table processing.

-   Copyright (C) 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2006, 2007 Free
-   Software Foundation, Inc.
+   Copyright (C) 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2006, 2007,
+   2009 Free Software Foundation, Inc.

    Written by Jim Meyering, 1992.

@@ -1012,7 +1012,11 @@ hash_delete (Hash_table *table, const void *entry)
 		 : (table->n_buckets * tuning->shrink_factor
 		    * tuning->growth_threshold));

-	      hash_rehash (table, candidate);
+	      if (hash_rehash (table, candidate))
+		{
+		  /* Failure to allocate memory in an attempt to
+		     shrink the table is not fatal.  */
+		}
 	    }
 	}
     }
